zulipterminal/ui_tools/boxes.py

```diff
@@ -130,24 +132,25 @@ def autocomplete_mentions(self, text: str, prefix_string: str
                  if match_user(user, text[len(prefix_string):])]
         user_typeahead = style_typeahead(users, prefix_string+'**{}**')
         combined_typeahead = group_typeahead + user_typeahead
+        combined_names = groups + users
 
-        return combined_typeahead
+        return combined_names, combined_typeahead
 
-    def autocomplete_streams(self, text: str) -> List[str]:
+    def autocomplete_streams(self, text: str) -> Tuple[List[str], List[str]]:
         streams_list = self.view.pinned_streams + self.view.unpinned_streams
         streams = [stream[0]
                    for stream in streams_list
                    if match_stream(stream, text[1:])]
         stream_typeahead = style_typeahead(streams, '#**{}**')
-        return stream_typeahead
+        return streams, stream_typeahead
```

Comment:
I wanted the `autocomplete_*` functions to return only the list of names, and do the styling in `generic_autocomplete`. But this becomes complicated for mentions, since groups and users have different styling

Reply:
I can understand this, but am wondering if instead of`Tuple[List[str], List[str]]` it might be cleaner with `List[Tuple[str, str]]`; then you would just return one list and index into the tuple? This means you can build it at the same time in the comprehensions.